presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/cli/EnvironmentUp.java

```diff
@@ -96,13 +116,38 @@ public void run()
             environment.start();
             log.info("Environment '%s' started", selectedEnvironmentProvider.getEnvironmentName());
 
+            if (background) {
+                killContainersReaperContainer();
+                return;
+            }
+
+            sleepUntilInterrupted();
+            log.info("Exiting, the containers will exit too");
+        }
+
+        private void killContainersReaperContainer()
+        {
             try (DockerClient dockerClient = DockerClientFactory.lazyClient()) {
                 log.info("Killing the testcontainers reaper container (Ryuk) so that environment can stay alive");
-                killContainersReaperContainer(dockerClient);
+                TestcontainersUtil.killContainersReaperContainer(dockerClient);
             }
             catch (IOException e) {
                 throw new UncheckedIOException(e);
             }
         }
+
+        private void sleepUntilInterrupted()
+        {
+            try {
+                //noinspection InfiniteLoopStatement
+                while (true) {
+                    Thread.sleep(Long.MAX_VALUE);
+                }
+            }
+            catch (InterruptedException e) {
+                log.info("Interrupted");
+                // It's OK not to restore interrupt flag here. When we return we're exiting the process.
```

Comment:
you didn't save much. Saving the flag would require less typing and nobody would ask for a comment ;)

Reply:
Right. But this wouldn't be really correct. We're responding to interrupt and no-one higher up the stack is going to respond.
So it would still require a comment :)